### PR TITLE
fix connection not release bug

### PR DESCRIPTION
**Pull request by ghjhhyuyuy**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Description:
When authenticate to graphd failed,the status of the connection has not been modified, resulting in the inability to continue using it in the future.I encountered this issue and thinking this commit can fix it.

## How do you solve it?
Modify connection status `is_used` to False

### CODE REVIEW

**Review by wey-gu — APPROVED** [review]
Great catch and fix, thanks for the contribution!

File: nebula3/gclient/net/Connection.py [hunk 1]
```diff
@@ -171,6 +171,8 @@ def authenticate(self, user_name, password):
         try:
             resp = self._connection.authenticate(user_name, password)
             if resp.error_code != ErrorCode.SUCCEEDED:
+                self._connection.is_used = False
```
Comment (HarrisChu) [inline nebula3/gclient/net/Connection.py:174]: the transport is not closed, prefer `self.close()`

File: nebula3/gclient/net/Connection.py [hunk 2]
```diff
@@ -171,6 +171,8 @@ def authenticate(self, user_name, password):
...
+                self._connection = None
```
Comment (Nicole00) [inline nebula3/gclient/net/Connection.py:175]: Authenticate failed does not mean the connection is broken, Why `set _connection = None`, we can just set `_connection.is_used` as `False`.

Let's see the connection pool how to call `authenticate` function:
```
    def get_session(self, user_name, password, retry_connect=True):
        """get session

        :param user_name: the user name to authenticate graphd
        :param password: the password to authenticate graphd
        :param retry_connect:
        :return: Session
        """
        connection = self.get_connection()
        if connection is None:
            raise NotValidConnectionException()
        try:
            auth_result = connection.authenticate(user_name, password)
            return Session(connection, auth_result, self, retry_connect)
        except Exception:
            raise
```
the auth operation is undependent with connection, so auth failed does not Influence the `connection` unless there are IO Error.

File: nebula3/gclient/net/Connection.py (hunk 1 above)
Comment (ghjhhyuyuy) [reply nebula3/gclient/net/Connection.py:174]: Is it necessary to close the connection here? I tend to release the connection back to the connection pool and wait for the next authenticate

File: nebula3/gclient/net/Connection.py (hunk 2 above)
Comment (ghjhhyuyuy) [reply nebula3/gclient/net/Connection.py:175]: I refer to the code under `release` function of `session.py`. I think you may right,  `self._connection.is_used = False` is  enought here

**Review by wey-gu — APPROVED** [review]

**Comment by wey-gu** [comment]
@HarrisChu @Nicole00 kindly check the change now, what do you think?

File: nebula3/gclient/net/Connection.py (hunk 2 above)
Comment (wey-gu) [reply nebula3/gclient/net/Connection.py]: Done as agreed.

**Pull request merged**